arcgis-ios-sdk-samples/Display information/Query features with Arcade expression/QueryFeaturesArcadeExpressionViewController.swift

```diff
@@ -0,0 +1,117 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class QueryFeaturesArcadeExpressionViewController: UIViewController {
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            mapView.map = makeMap()
+            // Set the touch delegate.
+            mapView.touchDelegate = self
+        }
+    }
+    
+    /// The arcade expression evaluation operation.
+    var evaluateOperation: AGSCancelable?
+    
+    /// Make and load a map.
+    func makeMap() -> AGSMap {
+        // Create a portal item with the portal and item ID.
+        let portalItem = AGSPortalItem(portal: .arcGISOnline(withLoginRequired: false), itemID: "14562fced3474190b52d315bc19127f6")
+        // Make a map with the portal item.
+        let map = AGSMap(item: portalItem)
+        // Load the map.
+        map.load { _ in
+            // Set the visibility of all but the RPD Beats layer to false.
+            map.operationalLayers.forEach { layer in
+                if let currentLayer = layer as? AGSLayer {
+                    currentLayer.isVisible = currentLayer.name.contains("RPD Beats")
+                }
+            }
+        }
+        return map
+    }
+    
+    /// Evaluate the arcade expression for the selected feature at the map point.
+    func evaluateArcadeInCallout(for feature: AGSArcGISFeature, at mapPoint: AGSPoint) {
+        // Show progress hud.
+        UIApplication.shared.showProgressHUD(message: "Evaluating")
+        // Instantiate a string containing the arcade expression.
+        let expressionValue =
+        """
+        var crimes = FeatureSetByName($map, 'Crime in the last 60 days');
+        return Count(Intersects($feature, crimes));
+        """
+        // Create an arcade expression using the string.
+        let expression = AGSArcadeExpression(expression: expressionValue)
+        // Create an arcade evaluator with the arcade expression and an arcade profile.
+        let evaluator = AGSArcadeEvaluator(expression: expression, profile: .formCalculation)
+        guard let map = mapView.map else { return }
+        let profileVariables = ["$feature": feature, "$map": map]
+        // Get the arcade evaluation result given the previously set profile variables.
+        if evaluateOperation != nil {
+            evaluateOperation?.cancel()
+        }
+        evaluateOperation = evaluator.evaluate(withProfileVariables: profileVariables) { [weak self] result, error in
+            // Dismiss progress hud.
+            UIApplication.shared.hideProgressHUD()
+            guard let self = self else { return }
+            if let result = result, let crimeCount = result.cast(to: .string) as? String {
```

Comment:
Can this line be combined with the previous `guard` statement

Reply:
It could, although I feel it may be better to keep separate. If I guard `result` and `crimeCount` as well, it would prevent the error from being presented.